test/conformance/kubeconform/link.go

```diff
@@ -19,71 +19,56 @@ package main
 import (
 	"fmt"
 	"io/ioutil"
-	"os"
-	"path/filepath"
-	"regexp"
 
 	"gopkg.in/yaml.v2"
 
 	"k8s.io/kubernetes/test/conformance/behaviors"
 )
 
 func link(o *options) error {
-	var behaviorFiles []string
 	behaviorsMapping := make(map[string][]string)
 	var conformanceDataList []behaviors.ConformanceData
 
-	err := filepath.Walk(o.behaviorsDir,
-		func(path string, info os.FileInfo, err error) error {
-			if err != nil {
-				return err
-			}
-			r, _ := regexp.Compile(".+.yaml$")
-			if r.MatchString(path) {
-				behaviorFiles = append(behaviorFiles, path)
-			}
-			return nil
-		})
+	behaviorFiles, err := behaviors.BehaviorFileList(o.behaviorsDir)
 	if err != nil {
 		return err
 	}
+
 	fmt.Println()
 	fmt.Printf("Using behaviors from these %d files:\n", len(behaviorFiles))
 	for _, f := range behaviorFiles {
 		fmt.Println("    ", f)
 	}
 	fmt.Println()
+
 	if o.listAll {
 		fmt.Println("All behaviors:")
 	} else {
 		fmt.Println("Behaviors not covered by any conformance test:")
 	}
 
 	for _, behaviorFile := range behaviorFiles {
-		var suite behaviors.Suite
-
-		yamlFile, err := ioutil.ReadFile(behaviorFile)
+		suite, err := behaviors.LoadSuite(behaviorFile)
 		if err != nil {
-			return err
+			return fmt.Errorf("%q", err.Error())
```

Comment:
Quick question: What's the benefit in wrapping the error like this?

Reply:
no benefit, thanks for catching this